### DISCUSSION TITLE
[IntelliSense] Add `tailwindFunctions` á la Tailwind's Prettier plugin

### DISCUSSION THREAD

**Discussion by happycollision** [tailwindlabs/tailwindcss#13036] [Ideas]
### Short summary via some imaginary settings

It would be so nice if the VSCode extension had the same `tailwindFunctions` feature that the Prettier plugin has so that we could get color swatches and definitions on hover for whatever strings we define within known functions.
Here's a potential setting config and the resulting behavior:

```js
// VSCode Settings JSON
{
...
"tailwindCSS.tailwindFunctions": ["tw"]
}
```
```js
// someJsFile
import tw from "./my-tailwind-helper"

const baseClassName = tw("flex flex-col")
// Hovering over "flex" above shows the definition in your IDE
```

### The long version

The combination of Tailwind, Tailwind's Prettier plugin, and Tailwind's VSCode extension are extraordinary. The Prettier plugin has this cool feature that allows you to define a `tailwindFunctions` array of strings which informs Prettier which functions in your code are expected to contain Tailwind classes so it knows which strings to rearrange. That is such a great feature.

I am sure you see where this is going: It would be so nice it the VSCode plugin had the same feature:

Hover a `className` attribute string (definition appears):
![CleanShot 2024-02-26 at 09 52 19@2x](https://github.com/tailwindlabs/tailwindcss/assets/3663628/83f94690-905b-4492-9dfc-8d45106ccbf2)

Hover a function with string args inside a `className` attribute (definition appears):
![CleanShot 2024-02-26 at 09 52 32@2x](https://github.com/tailwindlabs/tailwindcss/assets/3663628/6b96f664-d48d-4c95-aec8-0c8b0e939d84)

Hover a function registered with the Tailwind Prettier plugin (no class definition):
![CleanShot 2024-02-26 at 09 52 58@2x](https://github.com/tailwindlabs/tailwindcss/assets/3663628/5fdcd3cb-6dfa-4b0a-8277-5d8c90525c33)

Being able to get color swatches and definitions for arbitrary strings would be great!

**Comment by wongjn** [comment]
For what it's worth, you could try configuring the [`tailwindCSS.experimental.classRegex` setting](https://github.com/tailwindlabs/tailwindcss/issues/7553#issuecomment-735915659) to achieve something similar.
